core/opendaq/device/include/opendaq/connection_status_container_impl.h

```diff
@@ -0,0 +1,337 @@
+            statusesAliased[nameAlias] = this->statuses.get(connectionString);
+    }
+
+    *statuses = statusesAliased.detach();
+    return OPENDAQ_SUCCESS;
+}
+
+inline ErrCode ConnectionStatusContainerImpl::addConfigurationConnectionStatus(IString* connectionString, IEnumeration* initialValue)
+{
+    OPENDAQ_PARAM_NOT_NULL(connectionString);
+    OPENDAQ_PARAM_NOT_NULL(initialValue);
+
+    const auto connectionStringObj = StringPtr::Borrow(connectionString);
+    if (connectionStringObj == "")
+        return OPENDAQ_ERR_INVALIDPARAMETER;
+
+    std::scoped_lock lock(sync);
+
+    if (configConnectionStatusAdded || statuses.hasKey(connectionStringObj))
+        return OPENDAQ_ERR_ALREADYEXISTS;
+
+    statuses[connectionStringObj] = initialValue;
+    statusNameAliases[connectionStringObj] = ConfigurationConnectionStatusAlias;
+    configConnectionStatusAdded = true;
+
+    return OPENDAQ_SUCCESS;
+}
+
+inline ErrCode ConnectionStatusContainerImpl::addStreamingConnectionStatus(IString* connectionString, IEnumeration* initialValue, IBaseObject* streamingObject)
+{
+    OPENDAQ_PARAM_NOT_NULL(connectionString);
+    OPENDAQ_PARAM_NOT_NULL(initialValue);
+
+    const auto connectionStringObj = StringPtr::Borrow(connectionString);
+    if (connectionStringObj == "")
+        return OPENDAQ_ERR_INVALIDPARAMETER;
+
+    std::scoped_lock lock(sync);
+
+    if (statuses.hasKey(connectionStringObj))
+        return OPENDAQ_ERR_ALREADYEXISTS;
+
+    ++streamingConnectionsCounter;
+    statuses[connectionStringObj] = initialValue;
+    const StringPtr statusNameAlias = getStreamingStatusNameAlias(connectionStringObj);
+    statusNameAliases[connectionStringObj] = statusNameAlias;
+
+    if (triggerCoreEvent.assigned())
+    {
+        const CoreEventArgsPtr args = createWithImplementation<ICoreEventArgs, CoreEventArgsImpl>(
+            CoreEventId::ConnectionStatusChanged, Dict<IString, IBaseObject>({
+                                        {"StatusName", statusNameAlias},
+                                        {"StatusValue", initialValue},
+                                        {"ConnectionString", connectionStringObj},
+                                        {"ProtocolType", Integer((Int)ProtocolType::Streaming)},
+                                        {"StreamingObject", streamingObject}}));
+        triggerCoreEvent(args);
+    }
+
+    return OPENDAQ_SUCCESS;
+}
+
+inline ErrCode ConnectionStatusContainerImpl::removeStreamingConnectionStatus(IString* connectionString)
+{
+    OPENDAQ_PARAM_NOT_NULL(connectionString);
+
+    std::scoped_lock lock(sync);
+
+    if (!statuses.hasKey(connectionString))
+        return OPENDAQ_ERR_NOTFOUND;
+
+    const StringPtr statusNameAlias =
+        statusNameAliases.hasKey(connectionString)
+            ? statusNameAliases.remove(connectionString)
+            : nullptr;
+
+    auto value = statuses.remove(connectionString);
+    value = "NotAvailable";
```

Comment:
I'd rename this to "Removed", as the status no longer actually exists in the container.